mktestimg.c

```diff
@@ -22,87 +22,123 @@ enum {
-    printf("width and height have to be non-zero. channels can be 3 or 4.\n");
-    printf("available formats: png: %d, jpg: %d, bmp: %d\n", FORMAT_PNG, FORMAT_JPG, FORMAT_BMP);
+/**
+ * @brief Display the command-line manual.
+ *
+ * @param prog Executable name
+ */
+void usage(const char* prog) {
+    printf("Usage: %s [-v] -w <width> -h <height> [-f <format>] "
+           "[-c <channels>] [-o <output>]\n\n"
+           "  - <width> and <height> must be non-zero\n"
+           "  - <channels> can be 3 or 4\n"
+           "  - <formats> available:\n"
+           "      png: %d, jpg: %d, bmp: %d\n",
+           prog, FORMAT_PNG, FORMAT_JPG, FORMAT_BMP);
 }
 
-int main(int argc, char **argv) {
+int main(int argc, char** argv) {
+    if (argc == 2 && !strcmp(argv[1], "-v")) {
+        printf("%s v%s\n", argv[0], MKTESTIMG_VERSION_STR);
+        return EXIT_SUCCESS;
+    }
+
     if (argc < 3) {
         usage(argv[0]);
-        return 0;
+        return EXIT_FAILURE;
     }
 
-    char *output = NULL; 
+    char* output = NULL;
     int width = 0, height = 0, channels = 0, format = 0;
 
-    for (int i = 1; i < argc; i++) {
-        if (!strcmp(argv[i], "-v")) {
-            printf("%s v%s\n", argv[0], MKTESTIMG_VERSION_STR);
-            return 0;
-        }
-        else if (!strcmp(argv[i], "-w") && width == 0) {
+    for (int i = 1; i < argc; i++)
+        if (!strcmp(argv[i], "-w") && width == 0)
             width = atoi(argv[++i]);
-        }
-        else if (!strcmp(argv[i], "-h") && height == 0) {
+
+        else if (!strcmp(argv[i], "-h") && height == 0)
             height = atoi(argv[++i]);
-        }
-        else if (!strcmp(argv[i], "-f") && format == 0) {
+
+        else if (!strcmp(argv[i], "-f") && format == 0)
             format = atoi(argv[++i]);
-        }
-        else if (!strcmp(argv[i], "-c") && channels == 0) {
+
+        else if (!strcmp(argv[i], "-c") && channels == 0)
             channels = atoi(argv[++i]);
-        }
+
         else {
-            if (!i) continue;
+            if (!i)
+                continue;
             if (!output) {
-                if (!strcmp(argv[i], "-o")) i++;
-                output = copystr(argv[i]);
-            }
-            else {
-                printf("unknown argument: %s\n", argv[i]);
-                usage(argv[0]);
-                return 0;
+                if (!strcmp(argv[i], "-o"))
+                    i++;
+                output = copy_str(argv[i]);
+            } else {
+                printf("Unexpected field: %s\n"
+                       "Try typing: %s - to get help.\n",
+                       argv[i], argv[0]);
+                return EXIT_FAILURE;
             }
         }
+
+    if (width == 0 || height == 0 || format == 0 || channels == 0 ||
+        output == 0) {
+        fprintf(stderr, "error: Arguments are malformed.\n");
+        return EXIT_FAILURE;
```

Comment:
This check is completely unnecessary. Width and height are meant to be the only required arguments. All others are optional. The check for width and height is already done [here](https://github.com/iddev5/mktestimg/blob/master/mktestimg.c#L79). This check also renders line 108 to 111 (in your branch) useless, which isn't desired.

Reply:
The `atoi()` function returns zero when the conversion is unsuccessful. Using the logic-or statement, the code diagnoses whether one of the arguments was filled with malformed values or not. In this place, an edit is possible.